pkg/inventory/model/client.go

```diff
@@ -334,52 +336,25 @@ func (r *Client) Watch(model Model, handler EventHandler) (w *Watch, err error)
 			w = nil
 		}
 	}()
-	var itr fb.Iterator
-	func() {
-		r.dbMutex.RLock()
-		defer r.dbMutex.RUnlock()
-		itr, err = Table{r.db}.Iter(
-			model,
-			ListOptions{Detail: 1})
-	}()
-	if err != nil {
-		return
-	}
-	list := fb.NewList()
-	for {
-		model, hasNext, mErr := itr.Next()
-		if mErr != nil {
-			err = mErr
-			return
-		}
-		if !hasNext {
-			break
-		}
-		// Event.
-		event := &Event{
-			Action: Created,
-		}
-		mErr = list.Append(event)
-		if mErr != nil {
-			err = mErr
-			return
-		}
-		// Model.
-		mErr = list.Append(model)
-		if mErr != nil {
-			err = mErr
+	options := handler.Options()
+	var snapshot fb.Iterator
+	if options.Snapshot {
+		snapshot, err = Table{r.db}.Iter(model, ListOptions{Detail: 1})
+		if err != nil {
 			return
 		}
+	} else {
+		snapshot = &fb.EmptyIterator{}
 	}
-	w.notify(list.Iter())
-	w.Start()
+
+	w.Start(snapshot)
```

Comment:
Pass the (snapshot) list of models to the watch.
This is more explicit and better separation of concerns.  Mainly knowledge of now `Event` and model objects are written to the file-backed list.